tests/measure/editors/test_execution_editor.py

```diff
@@ -154,119 +157,139 @@ def test_model_observe_child_adding_removing(task):
     model._child_notifier_observer(notification)
 
 
-def test_execution_editor_widget(windows, task, dialog_sleep):
+def test_execution_editor_widget(windows, task, process_and_sleep):
     """Test the behavior of the execution editor widget.
 
     """
-    dialog_sleep = dialog_sleep or 1
     task.children[1].children[0].parallel = {}
 
     def get_task_widget(editor):
+        """Get the widget associated with the currently selected task.
+
+        """
         return editor.page_widget().widgets()[0].scroll_widget().widgets()[0]
 
     editor = ExecutionEditor(declaration=Editor(id='ecpy.execution_editor'),
                              selected_task=task)
     window = PageTestingWindow(widget=editor)
     window.show()
-    process_app_events()
-    sleep(dialog_sleep)
+    process_and_sleep()
 
+    # Select the complex task (ref ctask)
     ctask = task.children[1]
     editor.selected_task = ctask
-    process_app_events()
-    sleep(dialog_sleep)
+    process_and_sleep()
 
+    # Get the widget (ced) associated with ctask and alter the stoppable
+    # setting and the change propagation
     ced = get_task_widget(editor)
     ced.widgets()[0].checked = not ctask.stoppable
     process_app_events()
     assert ced.widgets()[0].checked == ctask.stoppable
-    sleep(dialog_sleep)
+    process_and_sleep()
 
+    # Set the complex task in parallel and check the update of the list of
+    # pools
     ctask.parallel['pool'] = 'test_'
     ced.widgets()[1].checked = True
-    process_app_events()
+    process_and_sleep()
     assert 'test_' in editor.pool_model.pools
-    sleep(dialog_sleep)
 
+    # Unset the wait setting, change the waiting pool, set the wait and check
+    # the list of pools
     ced.widgets()[3].checked = False
-    process_app_events()
-    sleep(dialog_sleep)
+    process_and_sleep()
     ctask.wait['no_wait'] = ['test2']
     ced.widgets()[3].checked = True
-    process_app_events()
+    process_and_sleep()
     assert 'test2' in editor.pool_model.pools
-    sleep(dialog_sleep)
 
+    # Change the selected pool for parallel and check the list of pools
     ced.widgets()[2].selected = 'test2'
-    process_app_events()
+    process_and_sleep()
     assert 'test' not in editor.pool_model.pools
-    sleep(dialog_sleep)
 
     def get_popup_content(parent):
         return parent.children[-1].central_widget().widgets()
 
+    # Create a new pool using the context menu on parallell
     ced.widgets()[2].children[0].children[0].triggered = True
-    process_app_events()
-    sleep(dialog_sleep)
+    process_and_sleep()
     process_app_events()  # So that the popup shows correctly
```

Comment:
Actually the showing of the popup takes several cycles and I need to add an additional sleep as it is the reasons for the failures of Travis.